prov/efa/src/efa_fabric.c

```diff
@@ -479,13 +492,40 @@ static int efa_get_gdr_support(char *device_name)
 		goto out;
 	}
 
-	ret = fi_read_file(get_sysfs_path(), gdr_path, &gdr_support, sizeof(char));
-	if (ret < 0)
+	ret = fi_read_file(get_sysfs_path(), gdr_path,
+			   gdr_support_str, MAX_GDR_SUPPORT_STRLEN-1);
+	if (ret < 0) {
+		ret = 0;
+		*gdr_support = 0;
+		goto out;
+	}
+
+	if (ret == 0) {
+		EFA_WARN(FI_LOG_FABRIC, "Sysfs file %s is empty\n", gdr_path);
+		ret = -FI_EINVAL;
+		goto out;
+	}
+
+	gdr_support_str[ret] = '\0';
+	*gdr_support = strtol(gdr_support_str, &endptr, 10);
+	if ((endptr - gdr_support_str) != 1) {
+		EFA_WARN(FI_LOG_FABRIC, "Unexpected sysfs file content! gdr_support_str: %s\n",
+			 gdr_support_str);
+		ret = -FI_EINVAL;
 		goto out;
+	}
 
-	ret = atoi(&gdr_support);
+	if (*gdr_support != 0 && *gdr_support != 1) {
+		EFA_WARN(FI_LOG_FABRIC, "Unexpected value of gdr_suport! gdr_support_str: %s gdr_support: %d\n",
+			 gdr_support_str, *gdr_support);
+		ret = -FI_EINVAL;
+		goto out;
+	}
+
+	ret = 0;
 out:
-	free(gdr_path);
+	if (gdr_path)
```

Comment:
This change isn't needed, free can handle NULL

Reply:
fixed in new revision.